python/cudf/utils/cudautils.py

```diff
@@ -795,6 +796,102 @@ def gpu_mark_seg_segments(begins, markers):
         markers[begins[i]] = 1
 
 
+# taken from the following repo:
+# https://github.com/numba/numba-examples/
+@cuda.jit
+def gpu_min_max(arr, min_max_array):
+    start = cuda.grid(1)
+    stride = cuda.gridsize(1)
+
+    # Array already seeded with starting values appropriate for x's dtype
+    # Not a problem if this array has already been updated
+    local_min = min_max_array[0]
+    local_max = min_max_array[1]
+
+    for i in range(start, arr.shape[0], stride):
+        element = arr[i]
+        local_min = min(element, local_min)
+        local_max = max(element, local_max)
+
+    # Now combine each thread local min and max
+    cuda.atomic.min(min_max_array, 0, local_min)
+    cuda.atomic.max(min_max_array, 1, local_max)
+
+
+def min_max(arr):
+    dtype_min, dtype_max = dtype_min_max(arr.dtype)
+    min_max_array_gpu = rmm.device_array(shape=(2,), dtype=arr.dtype)
+    min_max_array_gpu[0] = dtype_max
+    min_max_array_gpu[1] = dtype_min
+    gpu_min_max.forall(arr.size)(arr, min_max_array_gpu)
+    min_value = min_max_array_gpu[0]
+    max_value = min_max_array_gpu[1]
```

Comment:
This will trigger two device to host memory transfers separately, likely better to do it in one and then index into the small host array.

Reply:
This function is going away and will be replaced with just `column.max()` and `column.min()`.